atramhasis/scripts/generate_ldf_config.py

```diff
@@ -44,102 +43,104 @@ def main():
+            {"prefix": "hydra", "uri": "http://www.w3.org/ns/hydra/core#"},
+            {"prefix": "void", "uri": "http://rdfs.org/ns/void#"},
+            {"prefix": "skos", "uri": "http://www.w3.org/2004/02/skos/core#"},
+            {"prefix": "skos-thes", "uri": "http://purl.org/iso25964/skos-thes#"},
+        ],
     }
 
     if ldf_baseurl:
-        ldfconfig['baseURL'] = ldf_baseurl
+        ldfconfig["baseURL"] = ldf_baseurl
 
     if ldf_protocol:
-        ldfconfig['protocol'] = ldf_protocol
+        ldfconfig["protocol"] = ldf_protocol
 
     pids = []
     for p in skos_registry.get_providers():
         if any([not_shown in p.get_metadata()['subject'] for not_shown in ['external']]):
-            continue;
-        pid = p.get_metadata()['id']
+            continue
+        pid = p.get_metadata()["id"]
+        title = p.concept_scheme.label().label if p.concept_scheme.label() else pid
         pids.append(pid)
-        filename = os.path.join(dump_location, '%s-full' % pid)
-        filename_ttl = filename + '.ttl'
-        filename_hdt = filename + '.hdt'
-        if os.path.isfile(filename_hdt):
-            dumptype = 'HdtDatasource'
-            dumpfile = filename_hdt
-        else:
-            dumptype = 'TurtleDatasource'
-            dumpfile = filename_ttl
+        filename = os.path.join(dump_location, "%s-full" % pid)
+        dumptype = "HdtDatasource"
+        filetype = "hdtFile"
+        dumpfile = filename + ".hdt"
+
+        if not os.path.isfile(dumpfile):
+            dumptype = "TurtleDatasource"
+            filetype = "file"
+            dumpfile = filename + ".ttl"
+
         sourceconfig = {
-            'title': p.concept_scheme.label().label if p.concept_scheme.label() else pid,
-            'type': dumptype,
-            'settings': {
-                'file': dumpfile
-            }
+            "@id": f"urn:ldf-server:myDatasource{pid}",
+            "@type": dumptype,
+            "quads": False,  # TODO
+            "datasourcePath": pid,
+            "datasourceTitle": title,
+            filetype: dumpfile,
         }
+
         for n in p.concept_scheme.notes:
-            if n.type in ['definition', 'scopeNote']:
-                sourceconfig['description'] = n.note
+            if n.type in ["definition", "scopeNote"]:
+                sourceconfig["description"] = n.note
                 break
-        ldfconfig['datasources'][pid] = sourceconfig
+
+        ldfconfig["datasources"].append(sourceconfig)
 
     if len(pids):
-        sourceconfig = {
-            'title': 'All conceptschemes',
-            'type': 'CompositeDatasource',
-            'description': 'All conceptschemes contained in this Atramhasis instance together.',
-            'settings': {
-                'references': pids
-            }
+        composite_sourceconfig = {
+            "@id": "urn:ldf-server:myDatasourcecomposite",
+            "@type": "CompositeDatasource",
+            "quads": False,  # TODO
```

Comment:
@koenedaele @goessebr 
the [migration script](https://github.com/LinkedDataFragments/Server.js/wiki/Release-3.0.0#migration-from-2xx) explicitly set this value to False, so I did so as well here. Should quad patterns be supported? If this field is omitted from the config, it defaults to True